frontend/app/components/comment-form/text-expander.tsx

```diff
@@ -0,0 +1,102 @@
+/** @jsx createElement */
+import { createElement, RenderableProps, Fragment, render } from 'preact';
+import { StaticStore } from '@app/common/static_store';
+import { useEffect, useRef } from 'preact/hooks';
+import '@github/text-expander-element';
+import styles from './text-expander.module.pcss';
+import cx from 'classnames';
+import { Theme } from '@app/common/types';
+import useTheme from '@app/hooks/useTheme';
+
+type Emoji = {
+  key: string;
+  emoji: string;
+};
+
+function SuggestionList({ items, theme }: { items: Array<Emoji>; theme: Theme }) {
+  const isDarkTheme = theme === `dark`;
+  const suggesterClass = cx(styles.suggester, { [styles.suggesterDark]: isDarkTheme });
+  const suggesterItemClass = cx(styles.suggesterItem, { [styles.suggesterItemDark]: isDarkTheme });
+  return (
+    <ul className={suggesterClass}>
+      {items.map(({ key, emoji }: Emoji) => {
+        return (
+          // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
+          <li role="option" className={suggesterItemClass} data-value={key}>
+            <span className={styles.emojiResult}>{emoji}</span> {key}
+          </li>
+        );
+      })}
+    </ul>
+  );
+}
+
+function searchEmoji(key: string, text: string, theme: Theme) {
+  return import(/* webpackChunkName: "node-emoji" */ `node-emoji`)
+    .then(nodeEmoji => {
+      if (key === ':') {
+        const emojiList = nodeEmoji.search(text);
+        if (emojiList.length === 0) {
+          return Promise.resolve({ matched: false });
+        }
+        const fragment = document.createDocumentFragment();
+        render(<SuggestionList theme={theme} items={emojiList.slice(0, 5)} />, fragment);
```

Comment:
I think the first 5 emojis not relevant for daily use. I think we can choose "top 5 emojis" for first appear.
![Screenshot 2020-01-12 at 19 25 43](https://user-images.githubusercontent.com/2330682/72222030-54e39900-3571-11ea-8db8-556680edac28.png)

Reply:
It is good idea. It can be improved in the future. I don't know where I can this information, look like it depend on user.

Reply:
I found the rating, but I think we need to pick the most neutral of it. Like laughing, smiling, clapping, thumbs up or something like suggesting github/facebook for comment reaction.
https://home.unicode.org/emoji/emoji-frequency/

Reply:
Or it can be configurated :)

Reply:
just an idea: can we save the latest selections and display as defaults? As an initial, predefined default we can pick some from the list above.